app/main/presenters/search_presenters.py

```diff
@@ -97,6 +100,10 @@ def set_filter_states(filter_groups, request):
                     filter['value'] in param_values
                 )
 
+            filter_params = request_filters.copy()
+            filter_params.update({} if filter['checked'] else {filter['name']: filter['value']})
```

Comment:
I didn't quite have as much time as I'd like to review this but I wasn't quite clear about what this line is doing...